src/ast/passes/resource_analyser.h

```diff
@@ -21,7 +21,9 @@ namespace ast {
 // example the helper error metadata is still being collected during codegen.
 class ResourceAnalyser : public Visitor {
 public:
-  ResourceAnalyser(Node *root, std::ostream &out = std::cerr);
+  ResourceAnalyser(Node *root,
+                   BPFtrace &bpftrace,
```

Comment:
Could we just pass `Config` here?

Reply:
I went with that initially but that proved to be more of a pain for some reason. Also semantic_analyser has a bpftrace member var so I figured I'd stick with that pattern.

Reply:
Strange but ok. It's not that important that we should block the PR on it.